packages/core-transaction-pool/src/memory.ts

```diff
@@ -62,7 +62,7 @@ export class Memory {
             this.byExpirationIsSorted = true;
         }
 
-        const currentHeight: number = app.resolve("state").getLastBlock().data.height;
+        const currentHeight: number = app.resolve("blockchain").getLastHeight();
```

Comment:
Why did you suggest this change? Was there something wrong with `app.resolve("state").getLastBlock().data.height;`?

Your suggestion fails with:
`TypeError: core_container_1.app.resolve(...).getLastHeight is not a function`, same if `app.resolvePlugin<Blockchain.IBlockchain>("blockchain").getLastHeight()` is used instead.

Reply:
Just a shortcut for the same call but by using `getLastHeight` you won't be affected by changes to the underlying `state` storage as the return value will remain the same.

There is a small issue here with the startup of a node, you rely on bindings from `core-blockchain` but those won't be available until `core-blockchain` is mounted which happens a few plugins later. You would need a fallback to use the database on boot while `core-blockchain` bindings are not available.